zrml/orderbook/src/migrations.rs

```diff
@@ -251,20 +253,14 @@ mod tests {
     #[test]
     fn on_runtime_upgrade_is_noop_if_versions_are_not_correct() {
         ExtBuilder::default().build().execute_with(|| {
-            // Don't set up chain to signal that storage is already up to date.
-            let (_, new_orders) = construct_old_new_tuple();
-            populate_test_data::<Twox64Concat, OrderId, OrderOf<Runtime>>(
-                ORDER_BOOK,
-                ORDERS,
-                new_orders.clone(),
-            );
-            TranslateOrderStructure::<Runtime>::on_runtime_upgrade();
```

Comment:
You should leave the orders in to ensure they don't get messed up by the migration.

Reply:
I thought about this as I were making this change but came to the conclusion, that the `StorageVersion` would be changed in any case. So that's why I didn't care about the orders. If the storage root is the same before and after the migration and I know that the `StorageVersion` is changed in any case, I know that the orders are also not changed.